### DISCUSSION TITLE
sort on multiple params

### DISCUSSION THREAD

**Discussion by vorian77** [vincjo/datatables#76] [Q&A] [answered]
Am I correct to use **applySort** to apply multiple search params?

Given the following:

`sortItems = [ { orderBy: 'nameLast', direction: 'asc' }, { orderBy: 'nameFirst', direction: 'asc' } ]`

And code:

```
sortItems.forEach((so) => {
  handler.applySort({
    orderBy: so.orderBy,
    direction: so.direction
  })
})
```

The data table only sorts on the 'firstName' column. 

I'm trying to get it to sort "lastName" then "firstName".

**Comment by vincjo** [comment]
Your code is correct


I think the columns are sorted correctly in the order given in `sortItems`.
What can be confusing is that the "sorted column" state only applies to the last column `nameFirst`

**Comment by vorian77** [comment]
Hi @vincjo, thank you for your response and for building and maintaining this great tool!

I think I'm still missing something. Is the "sorted column" state the visual indicator? That visual indicator is specifying `nameFirst` is being applied, but the table is definitely sorted only on `nameFirst`. 

Here is the actual code and console log of `sortItems` being applied.

Can I somehow use .getSort() to determine exactly the current sort being applied to the table?

```
handler.clearSort()
if (sortItems) {
  sortItems.forEach((si: any) => {
    console.log('FormList.sortItem:', si)
    handler.applySort({
      orderBy: si._name,
      direction: si._codeDbListDir ? si._codeDbListDir : 'asc'
    })
  })
}
```

![Screenshot 2023-11-08 at 5 29 06 AM](https://github.com/vincjo/datatables/assets/1355098/93d78abf-9334-414b-b7ae-6f36f2e3e996)

**Comment by vincjo** [comment, marked as answer]
Thx
Yes i meant the visual indicator 👍 

I guess you expect last names to be sorted alphabetically. But if 2 people have the same, then the first names are sorted too, right?

Literally “sorting last names and first names”.

If you do it manually by clicking on the table, you must 1) click on `nameFirst` column, then 2) on `nameLast` to get this result.
Same way for applySort.
```ts
handler.sortAsc('nameFirst')
handler.sortAsc('nameLast')

// or
handler.applySort({ orderBy: 'nameFirst', direction: 'asc' })
handler.applySort({ orderBy: 'nameLast', direction: 'asc' })

// will sort items by nameLast and nameFirst
```
If i understood correctly you just need to reverse items in your `sortItems` array

**Reply by vorian77** [reply]
That was it, thanks again!

Do you have a Patreon account? (I'd love to support your work)

**Reply by vincjo** [reply]
Thank you very much for offering, i don't have a financing platform, it is not planned at the moment

**Comment by vincjo** [comment]
.getSort only returns the last sorted column.
The sort history is'nt exposed, but if there is a need I can add a method to provide it.

**Reply by vorian77** [reply]
No, an additional method is not necessary for me. I was just looking for more ways to debug my situation.
